src/build/GhosttyResources.zig

```diff
@@ -0,0 +1,257 @@
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Vim plugin
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("syntax/ghostty.vim", config_vim.syntax);
+        _ = wf.add("ftdetect/ghostty.vim", config_vim.ftdetect);
+        _ = wf.add("ftplugin/ghostty.vim", config_vim.ftplugin);
+        _ = wf.add("compiler/ghostty.vim", config_vim.compiler);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/vim/vimfiles",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Neovim plugin
+    // This is just a copy-paste of the Vim plugin, but using a Neovim subdir.
+    // By default, Neovim doesn't look inside share/vim/vimfiles. Some distros
+    // configure it to do that however. Fedora, does not as a counterexample.
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("syntax/ghostty.vim", config_vim.syntax);
+        _ = wf.add("ftdetect/ghostty.vim", config_vim.ftdetect);
+        _ = wf.add("ftplugin/ghostty.vim", config_vim.ftplugin);
+        _ = wf.add("compiler/ghostty.vim", config_vim.compiler);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/nvim/site",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // Sublime syntax highlighting for bat cli tool
+    // NOTE: The current implementation requires symlinking the generated
+    // 'ghostty.sublime-syntax' file from zig-out to the '~.config/bat/syntaxes'
+    // directory. The syntax then needs to be mapped to the correct language in
+    // the config file within the '~.config/bat' directory
+    // (ex: --map-syntax "/Users/user/.config/ghostty/config:Ghostty Config").
+    {
+        const wf = b.addWriteFiles();
+        _ = wf.add("ghostty.sublime-syntax", config_sublime_syntax.syntax);
+
+        const install_step = b.addInstallDirectory(.{
+            .source_dir = wf.getDirectory(),
+            .install_dir = .prefix,
+            .install_subdir = "share/bat/syntaxes",
+        });
+        try steps.append(&install_step.step);
+    }
+
+    // App (Linux)
+    if (cfg.target.result.os.tag == .linux) {
+        // https://developer.gnome.org/documentation/guidelines/maintainer/integrating.html
+
+        // Desktop file so that we have an icon and other metadata
+        try steps.append(&b.addInstallFile(
+            b.path("dist/linux/app.desktop"),
+            "share/applications/com.mitchellh.ghostty.desktop",
+        ).step);
+
+        // Right click menu action for Plasma desktop
+        try steps.append(&b.addInstallFile(
+            b.path("dist/linux/ghostty_dolphin.desktop"),
+            "share/kio/servicemenus/com.mitchellh.ghostty.desktop",
+        ).step);
```

Comment:
I wouldn't mind personally if this moved behind a build option as a GNOME user. Something like `-Dkde-dolphin-action=true` or something. Again, maybe it makes sense to default to true. Happy to turn it off for my own builds.

Reply:
Yeah I believe all this stuff should be defaulted to true, because when Ghostty is packaged, it could be used by any desktop environment downstream really.